### ISSUE TITLE
bug vibrantt egg

### ISSUE THREAD

**Issue by GoularPink** [malucooo/Otxserver-New#215]
Clica no egg (voce já tem a montaria), ai confirma em seguida, error:

Lua Script Error: [CreatureScript Interface]
data/creaturescripts/scripts/custom/VibrantEgg.lua:onModalWindow
data/creaturescripts/scripts/custom/VibrantEgg.lua:21: attempt to index a nil value
stack traceback:
        [C]: in function '__index'
        data/creaturescripts/scripts/custom/VibrantEgg.lua:21: in function <data/creaturescripts/scripts/custom/VibrantEgg.lua

obs: o vibrant egg vai sumir ainda (**erradamente**)

**Comment by Nicholasz23** [comment]
Na verdade o vibrant egg não funciona assim, o certo é clicar e ganhar a mount e não abrir ModalWindow.
Aqui segue um video de exemplo. 

[https://www.youtube.com/watch?v=c_fCSafPPtA](url)

**Comment by joseluis2g** [comment]
vibrant egg is just an action who gives you the mount if you completed the quest but I dont know what happens if you use it with no storages.

**Comment by GoularPink** [comment]
function onUse(cid, item, fromPosition, itemEx, toPosition, isHotkey)
    
	local player = Player(cid)
	
	if not player:hasMount(94) then
			player:addMount(t.id)
			player:getPosition():sendMagicEffect(15)
			item:remove(1)
			doCreatureSay(cid, "You receive the permission to ride a sparkion.", TALKTYPE_ORANGE_1)
            else
			player:getPosition():sendMagicEffect(3)
            player:sendTextMessage(MESSAGE_INFO_DESCR, "You already have this mount.")
	end
 
 
    return true
end

**Comment by joseluis2g** [comment]
mmm
```lua
function onUse(player, item, fromPosition, target, toPosition, isHotkey)
	local player = Player(cid)
	if player:getStorageValue(HEART_OF_DESTRUCTION_STORAGE) ~= 1 then
		if not player:hasMount(94) then
			player:addMount(94)
			player:getPosition():sendMagicEffect(CONST_ME_MAGIC_GREEN)
			player:say("You receive the permission to ride a sparkion", false, player, toPosition)
			player:setStorageValue(HEART_OF_DESTRUCTION_STORAGE, 1)
			item:remove(1)
		else
			player:sendCancelMessage('You have already obtained this mount!')
		end
	else
		return false
	end
	return true
end
```

**Comment by Eternal-Scripts** [comment]
```
function onUse(player, item, fromPosition, target, toPosition, isHotkey)
    if player:getStorageValue(HEART_OF_DESTRUCTION_STORAGE) ~= 1 then
        if not player:hasMount(94) then
            player:addMount(94)
            player:getPosition():sendMagicEffect(CONST_ME_MAGIC_GREEN)
            player:say("You receive the permission to ride a sparkion.", TALKTYPE_MONSTER_SAY, false, nil, player:getPosition())
            player:setStorageValue(HEART_OF_DESTRUCTION_STORAGE, 1)
            item:remove(1)
        else
            player:sendCancelMessage('You have already obtained this mount!')
        end
    end

    return true
end
```

**Comment by luanluciano93** [comment]
Fix #272 ?

**Comment by malucooo** [comment]
Fixed
